UI Customization document - initial chapters (#35579)
keycloak/keycloak: docs/documentation/ui_customization/topics/avatars.adoc

[[_avatars]]
== Using Avatars

> Both the admin and account consoles allow use of an avatar to personalize the user experience.  {project_name} supports avatars using the oidc standard `picture` claim.

Reviewer: ```suggestion
The Admin Console and Account Console both allow use of an avatar to personalize the user experience.  {project_name} supports avatars using the OpenID Connect standard `picture` claim.
```

> This `picture` claim should have a URI as its value.  The URI should point to the avatar meant to be displayed in the masthead section of the admin console or account console.

Reviewer: ```suggestion
The value for `picture` claim should be a URI that points to the avatar to be displayed in the masthead section of the console.
```

=== Setting a picture attribute from the admin console

> The simplest way to allow users to specify the avatar URI is to add a `picture` attribute in User profile.  Just go to the admin console and navigate to the `Realm Settings -> User profile` tab.

Reviewer: ```suggestion
A simple way to allow users to specify the avatar URI is to add a `picture` attribute in the User profile. In the Admin Console, click *Realm Settings* and the *User profile* tab.
```

Here is an example setup of the `picture` attribute:

> image::images/picture-attribute-general-settings.png[caption="",title="User profile picture attribute setup in admin console"]

Reviewer: .User profile picture attribute setup

=== Avatar success
Here is the result in account console once the URI is saved for the picture attribute:

> image::images/account-console-with-avatar.png[caption="",title="Avatar shown in account console"]

Reviewer: ```suggestion
.Avatar in Account Console
image::images/account-console-with-avatar.png[caption="",title="Avatar in Account Console"]
```

=== Important warning

> CAUTION: Allowing users to specify their own URI could lead to security concerns.  An avatar can contain malware so it is important that images come from a trusted source.  One way to accomplish this is to make sure that the `picture` has a regular expression validator to restrict the URI.

Reviewer: ```suggestion
CAUTION: Allowing users to specify their own URI could lead to security concerns.  An avatar can contain malware so ensure that images come from a trusted source.  One practical approach is to make sure that the `picture` has a regular expression validator to restrict the URI.
```
